core/include/mutex.h

```diff
@@ -106,6 +106,10 @@ static inline int mutex_trylock(mutex_t *mutex)
 /**
  * @brief Locks a mutex, blocking.
  *
+ * Mutexes are served in FIFO order. This ensures that, as long as no thread
```

Comment:
I think is is more confusing than correct.
Actually, mutex waiters are served sorted by priority (highest priority / lowest priority value first), then FIFO.

Reply:
Thanks for the clarification. I guess this confirms the behaviour is sufficiently complex as to need documentation.

Reply:
Yup. Maybe even some module-level overview documentation?